### DISCUSSION TITLE
Argo Rollouts using blue green configmap

### DISCUSSION THREAD

**Discussion by sourceful-karlson** [argoproj/argo-rollouts#1930] [Q&A] [answered]
Hi,

In our application, we would like to do hot swapping of a dataset.

This means that our blue green deployment, the only thing that we care about is change in configmap.

Is there a way in argo rollout for the blue to use a blue set of configmap and green to to use a green set of configmap?

**Comment by kostis-codefresh** [comment]
To the best of my knowledge this is not supported. Argo Rollouts references a deployment and not a configmap.

However you can easily do a workaround.

1. Deploy configmap1 and configmap2 on cluster (so both of them exist but with different name/values)
2. Create application container image v1  (that looks at configmap1) and container image v2 (that looks at configmap2)
3. Use Argo Rollouts normally for container images v1->v2

You could also have a single configmap which itself have v1-value and v2-value if you prefer that.

**Comment by sourceful-karlson** [comment]
Hi Kostis,

We have solve the problem by using a configmap generator that had the rollout using different configmap values

**Reply by kostis-codefresh** [reply]
great! maybe you can post a snippet of your solution here for other people to see?

**Reply by Dentrax** [reply]
Hey @sourceful-karlson, can you please enlighten us about your workaround? Thanks!

**Reply by sourceful-karlson** [reply]
https://github.com/kubernetes-sigs/kustomize/blob/master/examples/configGeneration.md

**Reply by Dentrax** [reply]
Thanks, that's cool! What about for CRDs? If I want to rollout (replace) CRDs or Secrets?
https://github.com/argoproj/argo-rollouts/discussions/2107 and https://github.com/argoproj/argo-rollouts/discussions/1346#discussioncomment-3002044

**Reply by sourceful-karlson** [reply]
No idea for them sorry, but Configmap should get you very far

**Reply by Dentrax** [reply]
Thanks. I'd be great if you could share a simple `Rollouts` resource example that i can combine with the kustomize config generator.

**Reply by showwin** [reply]
This document will help you. https://github.com/argoproj/argo-rollouts/blob/ddce67165b4832001d2ccb6d68798a9580ea9afe/docs/features/kustomize.md
Example code is here: https://github.com/argoproj/argo-rollouts/tree/ddce67165b4832001d2ccb6d68798a9580ea9afe/docs/features/kustomize/example.

**Reply by Dentrax** [reply]
Seems a bit hard and confusing to me. I still couldn't get the relation between Kustomization and ConfigMap/Secret/CRDs, what I want to do is _replacement_, not _patch_.

**Reply by kostis-codefresh** [reply, marked as answer]
@Dentrax I have written a guide for this problem https://codefresh.io/blog/progressive-delivery-for-kubernetes-config-maps-using-argo-rollouts/
